### PR TITLE
web/assets: types and initial tools for serving embedded assets via HTTP

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: web/assets/fs.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package assets
+
+import (
+	"io/fs"
+	"net/http"
+)
+
+// FS represents an HTTP capable file system.
+type FS interface {
```
Comment (amery) [inline web/assets/fs.go:9]: This doesn't belong in this commit

File: web/assets/checksum.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import (
+	"encoding/base64"
+	"io"
+
+	"lukechampine.com/blake3"
+)
+
+// BLAKE3Sum calculates a BLAKE3-256 checksum for the file,
+// rewinding the file afterwards.
+func BLAKE3Sum(file io.ReadSeeker) (string, error) {
```
Comment (amery) [inline web/assets/checksum.go:12]: io.Reader and optionally type check for Seeker?

**Comment by amery** [comment]
out of place `FS` interface removed and `BLAKE3Sum()` turned into `BLAKE3SumFile()` with optional rewind, if Seek() is implemented.

**Review by karasz — APPROVED** [review]
LGTM

**Comment by amery** [comment]
`darvaza.org/x/web@v0.4.0` tagged
